Applications/Spire/Include/Spire/Ui/MenuButton.hpp

```diff
@@ -24,6 +24,12 @@ namespace Spire {
       /** Returns the menu. */
       ContextMenu& get_menu();
 
+      /**
+       * Sets the message of the empty state.
+       * @param message The message displayed when the menu is empty.
+       */
+      void set_empty_message(const QString& message);
```

Comment:
Avoid setters.

Reply:
I once considered customizing the empty message through the constructor, but that approach isn’t very flexible. Functions that create specific `MenuButtons`, such as `make_menu_icon_button `and `make_menu_label_button`, would then need additional versions to support customizing the empty message.

Reply:
Okay thanks for letting me know.